core/chains/evm/gas/rollups/l1_oracle.go

```diff
@@ -49,10 +53,11 @@ func IsRollupWithL1Support(chainType chaintype.ChainType) bool {
 	return slices.Contains(supportedChainTypes, chainType)
 }
 
-func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle) (L1Oracle, error) {
+func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle, chainIDToClientMap map[string]EVMClient) (L1Oracle, error) {
```

Comment:
How about `<thing>`By`<index>` for the map?
```suggestion
func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle, clientsByChainID map[string]EVMClient) (L1Oracle, error) {
```